.github/scripts/prData.js

```diff
@@ -0,0 +1,112 @@
+// prData.js
+
+// Function to fetch pull requests associated with a commit
+async function fetchPullRequests(github, owner, repo, sha) {
+  return await github.rest.repos.listPullRequestsAssociatedWithCommit({
+    owner,
+    repo,
+    commit_sha: sha,
+  });
+}
+
+// Function to fetch the current release version
+async function getReleaseVersionValue(github, owner, repo) {
+  const { data } = await github.rest.actions.getRepoVariable({
+    owner,
+    repo,
+    name: "RELEASE_VERSION",
+  });
+  return data.value;
+}
+
+async function fetchReleaseBranchSha(github, owner, repo) {
+  const { data } = await github.rest.repos.getCommit({
+    owner,
+    repo,
+    ref: "heads/release",
+  });
+
+  if (data && data.sha) {
+    console.log("The release branch head SHA is: " + data.sha);
+    return data.sha;
+  } else {
+    throw new Error("No SHA found in the response");
+  }
+}
+
+// Function to process labels to determine new version and label
+function processLabelsAndVersion(labels, currentVersion) {
+  let versionParts = currentVersion.split(".").map((x) => parseInt(x, 10));
+  let appliedLabel;
+
+  if (labels.some((label) => label.name === "breaking-change")) {
+    versionParts[0] += 1;
+    versionParts[1] = 0;
+    versionParts[2] = 0;
+    appliedLabel = "breaking change";
+  } else if (
+    labels.some((label) => ["hotfix", "security", "bug"].includes(label.name))
+  ) {
+    versionParts[2] += 1;
+    appliedLabel = labels.find((label) =>
+      ["hotfix", "security", "bug"].includes(label.name),
+    ).name;
+  } else {
+    versionParts[1] += 1;
+    versionParts[2] = 0;
+    appliedLabel = labels.find((label) => label).name; // Catch-all increment
```

Comment:
Isn't this just taking the first label in the list?  If that's the case, there's no need to "find".